### PR TITLE
Edits Project Form 

### LINKED ISSUE #64: Backend functionality for supporting storage of images in project section.
This is the most updated version of the high-fidelity design on the Project Idea page. 
Screenshot 2024-12-14 at 2 38 41 PM

The frontend has been updated in #61 (as you can see below)
Screenshot 2024-12-14 at 2 39 38 PM

However, we have to now ensure that the backend can store files that are uploaded on the Project Idea Page. I am assigning this to the backend team.

### PR DESCRIPTION
**Pull request by FlaIespa**
## Description

This PR adds updates and enhancements to the Project Form to improve usability and functionality and hooks with Supabase to ensure that the files uploaded in the project are properly hooked with the backend. 

Closes #64  

Here are the changes introduced.
1. Added validation to ensure the following fields are mandatory, which means that the user cannot submit the form unless those fields are completed. 
- Project Name 
- Description (150 characters or less)
- Detailed Description of the Project
- Feedback Question
- Tags associated with the project

2. Added the section that handles the photos and videos upload to include three upload boxes in the same manner as the High Fidelity Design. So, when a file is uploaded the corresponding square fills with the uploaded image as a background, replacing the + icon. 

3. Addes a Snackbar notification for form validation errors to provide immediate feedback to users when they leave mandatory fields empty. 

4. Connected the section "Would you like to share any photos or videos?" with the backend storage. 

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Tests (adding or updating tests)
- [ ] Documentation update

## Testing Instructions
Checkout the branch `flavia/edit-project-form`

1. Validation Testing
- Try to submit the form with any of the mandatory fields left empty. Ensure that an error Snackbar appears. You can do that with all the fields that I listed as mandatory 
- Fill out all required fields and successfully submit the form.

2. Photos and Videos Section:
- Click on each dashed upload square and upload an image or video.
- Verify that the uploaded file is displayed within the respective square as a background image.
- Verify in Supabase if the file is properly saved with its unique folder inside the project-file bucket. 

4. Error Handling
- Upload invalid file types and verify that no visual or functional errors occur.
- Test for edge cases, such as attempting to upload more than three files.

## Screenshots (if applicable)

Submission Form 
Screenshot 2024-12-08 at 4 57 32 PM

Try Submission without filling out one of the mandatory fields
Screenshot 2024-12-08 at 4 57 55 PM

Submitting with all mandatory fields completed 
Unable to screenshot, because the load time to redirect to the "Welcome to the Platform Page is too fast" 

Project Assets are being fetched from the backend. 
Screenshot 2024-12-16 at 10 48 42 AM

### CODE REVIEW

File: src/components/ProjectIdeaForm.tsx [hunk 1]
```diff
@@ -39,6 +40,10 @@ const ProjectIdeaForm: React.FC<ProjectIdeaFormProps> = ({
 	const [projectDescription, setProjectDescription] = useState("");
 	const [feedbackQuestion, setFeedbackQuestion] = useState("");
 	const [selectedTags, setSelectedTags] = useState<string[]>([]);
+	// const [generalQuestion, setGeneralQuestion] = useState("");
+	const [uploads, setUploads] = useState<File[]>([]);
+	// const [additionalQuestion, setAdditionalQuestion] = useState("");
+	const [photosVideos, setPhotosVideos] = useState<File[]>([]);
```
Comment (lewiskyron) [inline src/components/ProjectIdeaForm.tsx:55]: remove these comments if they are not being used

Comment (FlaIespa) [reply src/components/ProjectIdeaForm.tsx:55]: removed
